core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -258,7 +258,11 @@ object RequestChannel extends Logging {
         m.responseSendTimeHist.update(Math.round(responseSendTimeMs))
         m.totalTimeHist.update(Math.round(totalTimeMs))
         m.requestBytesHist.update(sizeOfBodyInBytes)
-        m.messageConversionsTimeHist.foreach(_.update(Math.round(messageConversionsTimeMs)))
+        m.messageConversionsTimeHist.foreach(h => {
+          if (messageConversionsTimeMs > 0) {
```

Comment:
This is a kind of behavior change. Could you add comments for it? Also, we should have unit test to verify it

Reply:
Hi @chia7712, we can leave `messageConversionsTimeHist` update as usually. The `MessageConversionsTimeMs` is used when producer uses a older message format than the broker. The test case is used to test old message format, so I think we can just update the test case.